flake.nix

```diff
@@ -1,52 +1,68 @@
 {
   inputs = {
-    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
-    base16.url = "github:SenchoPens/base16.nix";
-
-    # Used for documentation
-    home-manager = {
-      url = "github:nix-community/home-manager";
-      inputs.nixpkgs.follows = "nixpkgs";
-    };
-
-    flake-compat = {
-      url = "github:edolstra/flake-compat";
-      flake = false;
-    };
-
-    # Templates
     base16-alacritty = {
-      url = "github:aarowill/base16-alacritty";
       flake = false;
+      url = "github:aarowill/base16-alacritty";
     };
+
     base16-alacritty-yaml = {
-      url = "github:aarowill/base16-alacritty/63d8ae5dfefe5db825dd4c699d0cdc2fc2c3eaf7";
       flake = false;
+      url = "github:aarowill/base16-alacritty/63d8ae5dfefe5db825dd4c699d0cdc2fc2c3eaf7";
     };
+
     base16-fish = {
-      url = "github:tomyun/base16-fish";
       flake = false;
+      url = "github:tomyun/base16-fish";
     };
+
     base16-foot = {
-      url = "github:tinted-theming/base16-foot";
       flake = false;
+      url = "github:tinted-theming/base16-foot";
     };
+
     base16-helix = {
+      flake = false;
       url = "github:tinted-theming/base16-helix";
+    };
+
+    base16-tmux = {
       flake = false;
+      url = "github:tinted-theming/base16-tmux";
     };
+
     base16-kitty = {
+      flake = false;
       url = "github:kdrag0n/base16-kitty";
+    };
+
+    base16-vim = {
       flake = false;
+      url = "github:chriskempson/base16-vim";
     };
-    base16-tmux = {
-      url = "github:tinted-theming/base16-tmux";
+
+    base16.url = "github:SenchoPens/base16.nix";
+
+    flake-compat = {
       flake = false;
+      url = "github:edolstra/flake-compat";
     };
-    base16-vim = {
-      url = "github:chriskempson/base16-vim";
+
+    gnome-shell = {
       flake = false;
+
+      # TODO: Unlocking the input and pointing to official repository requires
+      # updating the patch:
+      # https://github.com/danth/stylix/pull/224#discussion_r1460339607.
+      url = "github:GNOME/gnome-shell/45.1";
```

Comment:
We should use GitLab here since it's the original source (the GitHub version is a mirror).

Reply:
I am very well aware of the GitHub repository being a GitLab mirror: https://github.com/trueNAHO/stylix/commit/026bc2a2032ae309e0c3bfcf8c0b3ecb062a20c5. However, I do not know how to specify the revision for GitLab repositories. Note that `gitlab:GNOME/gnome-shell/45.1` does not work.

Reference: https://github.com/tweag/rfcs/blob/ca2845b81814c345945de4be0b36cd5fb4eb080b/rfcs/0049-flakes.md